File: 누적합/2559/main.py
n, k = map(int, input().split())
temp = list(map(int, input().split()))

ans = sum(temp[:k]) # 최대값(=정답) 변수
num = ans # 모든 구간 합을 체크할 변수
for i in range(n-k):
    num += (temp[i+k] - temp[i]) # num : i+1번째에서 시작하는 구간합 값
    if num > ans:
        ans = num
print(ans)

File: 누적합/2559/Main.java
import java.util.Scanner;

public class Main {
	public static void main(String[] args) {
		
        // 입력 받기
		Scanner sc = new Scanner(System.in);
	
		int n = sc.nextInt();
		int k = sc.nextInt();
		
		int[] temp = new int [n];
		for ( int i=0; i<n; i++) {
			temp[i] = sc.nextInt();
		}
		sc.close();
        
        // num, ans base case 생성
        int num = 0;
		for ( int i=0; i<k; i++) {
			num += temp[i];
		}
        int ans = num;

        // 최대 누적합 구하기
		for ( int i=0; i<n-k; i++) {
			num = num - temp[i] + temp[i+k];
			if (num > ans){
                ans = num;
            }
		}
		
		System.out.println(ans);
	}
}